### PR TITLE
feat: create user selection transaction

### PR DESCRIPTION
**Pull request by martinkaintas**

### CODE REVIEW

File: client/src/sdk/GameChannel.ts [hunk 1]
```diff
@@ -111,34 +121,70 @@ export class GameChannel {
...
     if (update?.op === 'OffChainNewContract') {
       const proposedBytecode = update.code;
-      await verifyContractBytecode(toRaw(this.sdk), proposedBytecode);
+      const isContractValid = await verifyContractBytecode(
+        this.getSdkWithoutProxy(),
+        proposedBytecode
+      );
+      popupData.title = 'Contract validation';
+      popupData.text = `Contract bytecode is 
+      ${isContractValid ? 'matcing' : 'not matching'}`;
```
Comment (Kalovelo) [inline client/src/sdk/GameChannel.ts:145]:
```suggestion
      ${isContractValid ? 'matching' : 'not matching'}`;
```

**Review by Kalovelo — CHANGES_REQUESTED** [review]
Looks great! Found only a simple typo.

I had a few suggestions which can be considered for implementation on another PR
### Disable cancel button in popup
Since we're going for the happy path, we could disable it for now and use a tooltip to explain the disabled state.
In this case, the `GenericButton` could be updated to include button props:
```
const props = defineProps<
  {
    text: string;
  } & ButtonHTMLAttributes
>();
``` 
However, Vue3 doesn't really support this yet (https://github.com/vuejs/core/issues/4294) yet there is a workaround: https://github.com/wheatjs/vite-plugin-vue-type-imports

### Popup button font weight
I guess this is based on the design files, but I think that the popup button font is too thin to be readable. We could increase it

File: client/src/sdk/GameChannel.ts [hunk 2]
```diff
@@ -111,34 +121,70 @@ export class GameChannel {
...
+        mainBtnAction: () => {
+          popupStore.resetPopUp();
+          resolve(this.getSdkWithoutProxy().signTransaction(tx, {}));
+          // call the callback if it exists
+          popupData?.mainBtnAction?.();
+        },
+        secBtnAction: () => {
+          popupStore.resetPopUp();
+          // @ts-expect-error reject the promise
+          resolve(1);
```
Comment (Kalovelo) [inline client/src/sdk/GameChannel.ts:175]:
```suggestion
    return new Promise((resolve, reject) => {
      popupStore.showPopUp({
        title: popupData?.title ?? `Signing ${tag}`,
        text: popupData?.text ?? 'Do you want to sign this transaction?',
        mainBtnText: popupData?.mainBtnText ?? 'Confirm',
        secBtnText: popupData?.secBtnText ?? 'Cancel',
        mainBtnAction: () => {
          popupStore.resetPopUp();
          resolve(this.getSdkWithoutProxy().signTransaction(tx, {}));
          // call the callback if it exists
          popupData?.mainBtnAction?.();
        },
        secBtnAction: () => {
          popupStore.resetPopUp();
          reject(1);
```

Not sure about this one's goal, but if it is to reject, `reject` function could be used instead

Comment (martinkaintas) [reply client/src/sdk/GameChannel.ts:175]: Even if we don't sign, we need to resolve the promise, there is no error happening, we just want to decline the tx. If we reject we get an error.

**Pull request merged**